### PR TITLE
Added route arrow drop in component.

### PR DESCRIPTION
**Pull request by cafesilencio**
### Description
Added route arrow drop in component.

### CODE REVIEW

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routearrow/DropInRouteArrow.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+import com.mapbox.navigation.ui.maps.route.arrow.api.MapboxRouteArrowApi
+import com.mapbox.navigation.ui.maps.route.arrow.api.MapboxRouteArrowView
+import com.mapbox.navigation.ui.maps.route.arrow.model.RouteArrowOptions
+import com.mapbox.navigation.utils.internal.ifNonNull
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+internal class DropInRouteArrow(
```
Comment (kmadsen) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routearrow/DropInRouteArrow.kt:14]: we're looking to get away from `DropIn*`

This looks like a `RouteArrowComponent`

File: libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/component/routearrow/DropInRouteArrowTest.kt [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.slot
+import io.mockk.unmockkStatic
+import io.mockk.verify
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+
+class DropInRouteArrowTest {
```
Comment (kmadsen) [inline libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/component/routearrow/DropInRouteArrowTest.kt:26]: 💯

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationView.kt [hunk 3]
```diff
@@ -80,6 +81,16 @@ class DropInNavigationView @JvmOverloads constructor(
...
     }
 
+    /**
+     * Provide custom route arrow options to override the default options. This must be called
+     * before your activity or fragment's onStart() in order to take effect.
+     *
+     * @param options the [RouteArrowOptions] to use.
+     */
+    fun customize(options: RouteArrowOptions) {
+        navigationContext.routeArrowOptions = options
```
Comment (kmadsen) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationView.kt:91]: (out of scope note)

Something that would be good to think about, is if we can make a mechanism that allows up you update these options at runtime. For example, could a developer change the route arrow at runtime?

Comment (abhishek1508) [reply libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationView.kt:91]: A good chunk of this question will be answered once we have NSS prototype ready

**Review by abhishek1508 — APPROVED** [review]

**Pull request merged**